terraform/modules/open_id_connect/policies.tf

```diff
@@ -0,0 +1,76 @@
+data "aws_iam_policy_document" "github_actions_terraform_plan_assume_role" {
+  statement {
+    actions = ["sts:AssumeRoleWithWebIdentity"]
+
+    principals {
+      type        = "Federated"
+      identifiers = [aws_iam_openid_connect_provider.github.arn]
+    }
+
+    condition {
+      test     = "StringEquals"
+      values   = ["sts.amazonaws.com"]
+      variable = "token.actions.githubusercontent.com:aud"
+    }
+
+    condition {
+      test = "StringLike"
+      values = [
+        "repo:communitiesuk/delta-common-infrastructure:*"
+      ]
+      variable = "token.actions.githubusercontent.com:sub"
+    }
+  }
+}
+
+data "aws_iam_policy" "terraform_state_read_only" {
+  arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:policy/tf-state-read-only"
+}
+
+resource "aws_iam_role_policy_attachment" "github_actions_terraform_plan_state_read" {
+  role       = aws_iam_role.github_actions_terraform_plan.name
+  policy_arn = data.aws_iam_policy.terraform_state_read_only.arn
+}
+
+data "aws_iam_policy" "read_only_access" {
+  arn = "arn:aws:iam::aws:policy/ReadOnlyAccess"
+}
+
+resource "aws_iam_role_policy_attachment" "github_actions_terraform_plan_read_only_access" {
+  role       = aws_iam_role.github_actions_terraform_plan.name
+  policy_arn = data.aws_iam_policy.read_only_access.arn
+}
+
+data "aws_iam_policy" "administrator_access" {
+  arn = "arn:aws:iam::aws:policy/AdministratorAccess"
+}
+
+resource "aws_iam_role_policy_attachment" "github_actions_terraform_admin_access" {
+  role       = aws_iam_role.github_actions_terraform_admin.name
+  policy_arn = data.aws_iam_policy.administrator_access.arn
+}
+
+data "aws_iam_policy_document" "github_actions_terraform_admin_assume_role" {
+  statement {
+    actions = ["sts:AssumeRoleWithWebIdentity"]
+
+    principals {
+      type        = "Federated"
+      identifiers = [aws_iam_openid_connect_provider.github.arn]
+    }
+
+    condition {
+      test     = "StringEquals"
+      values   = ["sts.amazonaws.com"]
+      variable = "token.actions.githubusercontent.com:aud"
+    }
+
+    condition {
+      test = "StringEquals"
+      values = [
+        "repo:communitiesuk/delta-common-infrastructure:environment:${var.environment}"
```

Comment:
if I remove the default value, how do I set it for the staging main.tf file (where it wont be used)

Reply:
not sure I quite understand ..

Reply:
This module is only referenced from test/main.tf now, so it will only allow `repo:communitiesuk/delta-common-infrastructure:environment:test` and not `:staging`. We don't use it for staging in practice anyway so we should remove it from the GitHub config https://github.com/communitiesuk/delta-common-infrastructure/settings/environments

Reply:
ah okay, understood. I'm not able to access that URL, assuming its because I'm not admin in common-infra